File: src/main/java/com/mparticle/kits/ButtonMerchantWrapper.java
package com.mparticle.kits;

import android.content.Context;
import android.content.Intent;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.usebutton.merchant.ButtonMerchant;
import com.usebutton.merchant.ButtonMerchant.AttributionTokenListener;
import com.usebutton.merchant.ButtonProductCompatible;
import com.usebutton.merchant.PostInstallIntentListener;

import java.util.List;

/**
 * Wrapper class for {@link ButtonMerchant} to allow for testing the library's static methods
 */
public class ButtonMerchantWrapper {

    public void configure(@NonNull Context context, @NonNull String applicationId) {
        ButtonMerchant.configure(context, applicationId);
    }

    public void trackIncomingIntent(@NonNull Context context, @NonNull Intent intent) {
        ButtonMerchant.trackIncomingIntent(context, intent);
    }

    public String getAttributionToken(@NonNull Context context) {
        return ButtonMerchant.getAttributionToken(context);
    }

    public void addAttributionTokenListener(@NonNull Context context, @NonNull
            AttributionTokenListener listener) {
        ButtonMerchant.addAttributionTokenListener(context, listener);
    }

    public void handlePostInstallIntent(@NonNull Context context, @NonNull
            PostInstallIntentListener listener) {
        ButtonMerchant.handlePostInstallIntent(context, listener);
    }

    public void clearAllData(@NonNull Context context) {
        ButtonMerchant.clearAllData(context);
    }

    public void trackProductViewed(@Nullable ButtonProductCompatible product) {
        ButtonMerchant.activity().productViewed(product);
    }

    public void trackAddToCart(@Nullable ButtonProductCompatible product) {
        ButtonMerchant.activity().productAddedToCart(product);
    }

    public void trackCartViewed(@NonNull List<ButtonProductCompatible> products) {
        ButtonMerchant.activity().cartViewed(products);
    }
}

File: src/main/kotlin/com/mparticle/kits/ButtonMerchantWrapper.kt
package com.mparticle.kits

import android.content.Context
import com.usebutton.merchant.ButtonMerchant
import android.content.Intent
import com.usebutton.merchant.ButtonMerchant.AttributionTokenListener
import com.usebutton.merchant.PostInstallIntentListener
import com.usebutton.merchant.ButtonProductCompatible

/**
 * Wrapper class for [ButtonMerchant] to allow for testing the library's static methods
 */
class ButtonMerchantWrapper {
    fun configure(context: Context, applicationId: String) {
        ButtonMerchant.configure(context, applicationId)
    }

    fun trackIncomingIntent(context: Context, intent: Intent) {
        ButtonMerchant.trackIncomingIntent(context, intent)
    }

    fun getAttributionToken(context: Context): String? {
        return ButtonMerchant.getAttributionToken(context)
    }

    fun addAttributionTokenListener(context: Context, listener: AttributionTokenListener) {
        ButtonMerchant.addAttributionTokenListener(context, listener)
    }

    fun handlePostInstallIntent(context: Context, listener: PostInstallIntentListener) {
        ButtonMerchant.handlePostInstallIntent(context, listener)
    }

    fun clearAllData(context: Context) {
        ButtonMerchant.clearAllData(context)
    }

    fun trackProductViewed(product: ButtonProductCompatible?) {
        ButtonMerchant.activity().productViewed(product)
    }

    fun trackAddToCart(product: ButtonProductCompatible?) {
        ButtonMerchant.activity().productAddedToCart(product)
    }

    fun trackCartViewed(products: List<ButtonProductCompatible?>) {
        ButtonMerchant.activity().cartViewed(products)
    }
}